ocp_resources/resource.py

```diff
@@ -1171,21 +1209,18 @@ def keys_to_hash(self) -> List[str]:
 
          Example:
              given a dict: {"spec": {"data": <value_to_hash>}}
-             To hash spec['data'] key pass: ["spec..data"]
+             To hash spec['data'] key pass: ["spec>data"]
         """
         return []
 
     def hash_resource_dict(self, resource_dict: Dict[Any, Any]) -> Dict[Any, Any]:
+        if not isinstance(resource_dict, dict):
+            raise ValueError("Expected a dictionary as the first argument")
```

Comment:
why is this needed? we do not check types

Reply:
@myakove suggested adding this one.

Reply:
It's better to have a nice error when failing, typing is not enforced, only suggests the type (by IDE)

Reply:
but we do not have such checks in the wrapper

Reply:
Let me know how do you want me to proceed.

Reply:
@myakove / @rnetser please let me know how you want me to proceed here. I will update accordingly.

Reply:
leave the error with the msg, we should fail nice